examples/hosts/app-integration/external-controller/src/view.ts

```diff
@@ -57,10 +58,20 @@ export function renderAudience(audience: IFrsAudience, div: HTMLDivElement) {
         const members = audience.getMembers();
         const self = audience.getMyself();
         const memberNames: string[] = [];
+        const useFrs = process.env.FLUID_CLIENT === "frs";
+
         members.forEach((member) => {
             if (member.userId !== self?.userId) {
                 memberNames.push(member.userName);
             }
+
+            const additionalDetails = useFrs ? member.additionalDetails as ICustomUserDetails : {};
+            if(Object.keys(additionalDetails).length !== 0) {
+                for(const [key, value] of Object.entries(additionalDetails)) {
+                    const keyName = key.charAt(0).toUpperCase().concat(key.substr(1));
+                    memberNames.push(keyName.concat(": ").concat(value as string));
```

Comment:
The iteration here doesn't look quite right, `memberNames` is now holding an array of mixed strings: usernames, genders, and emails.

I'd recommend changing the array to instead hold all the strings constructed for each member.
You can cast `member` on line 63 to be `FrsMember<ICustomUserDetails>` and decide using `useFrs` on whether this member's string will just contain their username or the username and `additionalDetails` values combined

i.e. perhaps something like this:
```
members.forEach((member: FrsMember<ICustomUserDetails>) => {
            if (member.userId !== self?.userId) {
                if (useFrs) {
                    const memberString = `${member.userName}: {Gender: ${member.additionalDetails?.gender},
                        Email: ${member.additionalDetails?.email}}`;
                    memberStrings.push(memberString);
                } else {
                    memberStrings.push(member.userName);
                }
            }
        });
audienceDiv.innerHTML = `
    Current User: ${self?.userName} <br />
    Other Users: ${memberStrings.join(", ")}
`;
```

Reply:
memberStrings would be pushed to memberNames instead?

Reply:
memberStrings would replace memberNames, since we aren't just rendering names any longer 😄